tests/test_mcm.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+
+from flight.apps.adcs.mcm import (
+    ControllerHandler,
+    MagneticCoilAllocator,
+    get_spin_stabilizing_dipole_moment,
+    get_sun_pointing_dipole_moment,
+)
+from ulab import numpy as np
+
+
+@pytest.fixture
+def readings() -> dict:
+    sun_vector = np.random.uniform(size=3)
+    magnetic_field = np.random.uniform(size=3)
+    spin_error = np.random.uniform(size=3)
+    pointing_error = np.random.uniform(size=3)
+    dipole_moment = np.random.uniform(size=3)
+    return {
+        "sun_vector": sun_vector,
+        "magnetic_field": magnetic_field,
+        "spin_error": spin_error,
+        "pointing_error": pointing_error,
+        "dipole_moment": dipole_moment,
+    }
+
+
+def test_spin_stabilizing(readings) -> None:
+    dipole_moment = get_spin_stabilizing_dipole_moment(
+        readings["magnetic_field"],
+        readings["spin_error"],
+    )
+    assert dipole_moment.shape == (3,)
+
+
+def test_sun_pointing(readings) -> None:
+    dipole_moment = get_sun_pointing_dipole_moment(
+        readings["sun_vector"],
+        readings["pointing_error"],
+    )
+    assert dipole_moment.shape == (3,)
```

Comment:
Here, test the shape output is good but you want to test the computation as well with a clear input value (instead of random) and compare it to your reference output to make sure the computation is correct. 

You want also to test nominal cases and corner cases (near zero) or when a zero norm appears somewhere.

Be paranoid